File: solutions/java/src/main/java/playground/algorithm/ReverseInteger.java
package playground.algorithm;

public class ReverseInteger {

    public int reverse(int x) {
        int n = x;
        long result = 0;
        while (n != 0) {
            result = result * 10 + n % 10;
            n /= 10;
        }

        if (result > Integer.MAX_VALUE || result < Integer.MIN_VALUE) {
            return 0;
        }

        return (int) result;
    }
}

File: solutions/go/src/playground/algorithm/reverse_integer.go
package algorithm

func reverseInteger(x int) int {
	result := int64(0)
	for x != 0 {
		result = result*10 + int64(x)%10
		x /= 10
	}

	if result > 2147483647 || result < -2147483648 {
		return 0
	}

	return int(result)
}
